src/openrct2/util/util.h

```diff
@@ -56,4 +56,12 @@ uint32 util_rand();
 uint8 *util_zlib_deflate(uint8 *data, size_t data_in_size, size_t *data_out_size);
 uint8 *util_zlib_inflate(uint8 *data, size_t data_in_size, size_t *data_out_size);
 
+#define add_clamp(value, value_to_add, min_cap, max_cap) \
+	if ((max_cap - abs(value_to_add)) < abs(value) && (value < 0) == (value_to_add < 0)) { \
```

Comment:
I'd suggest something like this:

```
#define add_clamp(value, value_to_add, min_cap, max_cap) \
	if (((value_to_add > 0) && (value > (INT_MAX - value_to_add))) { \
		value = max_cap; \
    } \
    else if ((value_to_add < 0) && (value < (INT_MIN - value_to_add)))) { \
        value = min_cap; \
    } \
	else { \
		value += value_to_add; \
	} \
}
```

It's untested, but afapted from here: https://www.securecoding.cert.org/confluence/display/c/INT32-C.+Ensure+that+operations+on+signed+integers+do+not+result+in+overflow